docs/src/use/core-concepts/glossary.md

```diff
@@ -0,0 +1,397 @@
+
+## P
+
+### Parser
+
+A package that reads in a string and converts it to an [AST](#abstract-syntax-tree-ast).
+
+By default, ESLint uses [Espree](https://github.com/eslint/espree) parser, which is compatible with standard JavaScript runtimes and versions.
+
+Custom parsers let ESLint parse non-standard JavaScript syntax.
+Often custom parsers are included as part of shareable configurations or plugins, so you don’t have to use them directly.
+For example, [@typescript-eslint/parser](https://www.npmjs.com/package/@typescript-eslint/parser) is a custom parser included in the [typescript-eslint](https://typescript-eslint.io) project that lets ESLint parse TypeScript code.
+
+For more information on using parsers with ESLint, see [Configure a Parser](../configure/parser).
+
+### Plugin
+
+A package that can contain a set of [configurations](#shareable-config-configuration), [environments](#environment), [processors](#processor), and/or [rules](#rule).
+
+A popular use case for plugins is to enforce best practices for a framework.
+For example, [@angular-eslint/eslint-plugin](https://www.npmjs.com/package/@angular-eslint/eslint-plugin) contains best practices for using the Angular framework.
+
+For more information, refer to [Configure Plugins](../configure/plugins).
+
+### Processor
+
+A part of a plugin that extracts JavaScript code from other kinds of files, then lets ESLint lint the JavaScript code.
+Alternatively, processors can convert JavaScript code during preprocessing.
+
+For example, [`eslint-plugin-markdown`](https://github.com/eslint/eslint-plugin-markdown) includes a processor that converts the text of <code>```</code> code blocks in Markdown files into code that can be linted.
+
+For more information on configuring parsers, see [Plugins > Specify a Processor](../configure/plugins#specify-a-processor).
+
+## R
+
+### Report
+
+A collection of [violations](#violation) from a single ESLint run.
+
+When ESLint runs on source files, it will pass an [AST](#abstract-syntax-tree-ast) for each source file to each configured [rule](#rule).
+The collection of violations from each of the rules will be packaged together and passed to a [formatter](#formatter-linting) to be presented to the user.
+
+### Rule
+
+A piece of logic that checks [AST](#abstract-syntax-tree-ast) for the presence of a code defect.
+
+ESLint provides a large collections of rules that check for common JavaScript code issues.
+Many more lint rules may be loaded in by [plugins](#plugin).
+
+For an overview of rules provided, see [Core Concepts > Rules](../core-concepts#rules).
+
+## S
+
+### Selector
+
+A description of how to search for [nodes](#node) within an [AST](#abstract-syntax-tree-ast).
+
+ESLint [rules](#rule) use [ESQuery](#esquery) selectors to find nodes that should be checked.
+
+### Severity
+
+What level of reporting a rule is configured to run, if at all.
+
+ESLint supports three levels of severity:
```

Comment:
Worth mentioning the numeric equivalents?

Reply:
Aside: I have never seen a use case for those where I've liked the brevity more than I've disliked the lack of clarity...